### ISSUE TITLE
Error: TypeError: Cannot read property 'match' of undefined

### ISSUE THREAD

**Issue by haps-basset** [node-red/node-red-nodegen#99]
Hello 
I cannot find solution to the following problem, I am trying to generate a node-red template from a function file called template.js following your “how to”  but I am getting the following error message:
Error: TypeError: Cannot read property 'match' of undefined
I am using node 8.16 and node-red 0.20.60
The content of template.js is 
“//
return msg;
As I  have said, I was not able any workable solution on internet therefore I would appreciate if you could suggest some workable solutions or troubleshooting procedures.
Haps

**Comment by kazuhitoyokoi** [comment]
Thank you for raising the issue. I think that the invalid file format in the source file caused the error.
The source file needs to contain `name` and `output` in the comments at the first and second lines respectively as the following example.

```
$ cat ~/.node-red/lib/functions/lowercase.js
// name: lowercase
// outputs: 1
msg.payload = msg.payload.toLowerCase();
return msg;
```

The source file can be generated automatically using the library menu in function node property UI.
Could you check your source file?

**Comment by kazuhitoyokoi** [comment]
@haps-basset Can I close this issue? I think that you can successfully generate the node using Node generator if the source file exported from the library menu on the function node is used.

**Comment by haps-basset** [comment]
Hello
Thank you for effort in this matter, much appreciated.  Please close the call
Haps

**Comment by kazuhitoyokoi** [comment]
@haps-basset Thank you for your reply. I'll close it. :)

**Comment by juggledad** [comment]
This issue will continue to come up until the documation is fixed. 

In example 2 it says:

> On Node-RED flow editor, save function node to library with file name (lower-case.js).

But currently, saving a node to the library saves it in json format. If you name it 'lower-case.js' it will be saved as 'lower-case.js.json' and be in json format

Maybe I'm missing something on the export step but I found I had to copy the contents of the function node and in a text editor create the 'lower-case.js' file with the extra first two lines containing the name and outputs.
